### PR TITLE
Wishlist cypress test

### PR DESCRIPTION
**Pull request by dpatil-magento**
## Description

This is E2E test for single wishlist. Also included Category feature which is in progress.

## Related Issue

Closes [PWA-1677](https://jira.corp.magento.com/browse/PWA-1677).

### Verification Steps

1. Checkout this branch locally and update test url to https://develop.pwa-venia.com/
2. `cd venia-integration-test`
2. `yarn install && yarn test`
## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: venia-integration-tests/src/fixtures/accountAccess/index.js
Comment (fooman) [inline venia-integration-tests/src/fixtures/accountAccess/index.js:10]:
```suggestion
export const accountEmail = `${id}test@example.com`
```
Best to take one of these https://tools.ietf.org/html/rfc2606

Comment (dpatil-magento) [reply venia-integration-tests/src/fixtures/accountAccess/index.js]: thanks for the feedback @fooman , updated.

File: venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js
Comment (revanth0212) [inline venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js:34]: I understand why you went for the re-usable function but I feel we should assert in the test instead of using it as a common utility. That's how we do it in the jest tests as well. The reason being, when someone is reading the test, the ultimate thing a test is gonna prove is the assertion and they would like to read the assertion in the test. It gets hard if we don't know what the assertion does.

Thats my view. Thoughts guys?

File: venia-integration-tests/src/support/visitPageCommand.js
Comment (fooman) [inline venia-integration-tests/src/support/visitPageCommand.js:1]: I am not convinced that this approach is going to match end user behaviour. An end user will click themselves through the react app to navigate. This has a different behaviour to typing the url in the address bar.  So each test should at max have 1 visitPage use at the very start as the entrypoint.

Additionally I suggest removing the hard coded wait. From experience I can say that some will require a lot more than 5000, depending on back-end. It would be best to wait for the appropriate spinning loader to no longer be present or the next element required for the next interaction being there.

Comment (dpatil-magento) [reply venia-integration-tests/src/support/visitPageCommand.js:1]: Yeah agree on that point @fooman however these direct page visit would be used to avoid retesting already tested user flow part of other tests which makes test more lighter, stable and decoupled, examples -
1. Add product to cart - Actual user navigation to PDP would be part of category listing test so when need to add product to cart we would direct hit pdp.
2. Order placement flows - we will use user navigation to `cart` `checkout` pages in 1 or 2 tests but when not needed to check link functionality `Edit Shipping Bag`  `checkout` or `go to checkout` but still need to test something on checkout flows then will hit direct checkout.
3. Actual user category visits would be part of Mega menu tests but rest of the times we will directly go to needed category.

Regrading 5 sec wait - currently we dont have better way to handle full page load so might be temporary until we find better solution. One example - When we visit home page and try to create account >> cypress automatically waits and clicks on `Sign In`  button but at this time the page is still loading. Now when page loads the signin dropdown is collapsed, cypress does not find `Create an Account` button and fail the test.  In this case we did try to wait for home page banner image to load but its not same banner image always, it keeps changing and test fails randomly.

File: venia-integration-tests/src/actions/wishlist/index.js
Comment (revanth0212) [inline venia-integration-tests/src/actions/wishlist/index.js:17]: I am thinking about moving these assertions to a folder of their own. Ill make the changes and push the commit.

**Review by revanth0212 — APPROVED** [review]

**Review by revanth0212 — APPROVED** [review]

File: venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js
Comment (sirugh) [inline venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js:44]: Is this something that should get done before we merge this? If not, can you open a ticket and add to the comment?

File: venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js
Comment (sirugh) [reply venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js:57]: We discussed in chat but I don't know if that was ever applied here - I think we said to go with functions for non assertions, such as `navigateToCart` and `addItemToCart` but to use assertions for things like `didAddItemToCart`

File: venia-integration-tests/src/assertions/wishlist/index.js
Comment (sirugh) [inline venia-integration-tests/src/assertions/wishlist/index.js:46]: [Image from Gyazo](https://gyazo.com/7551fbd0aa603a6cbee2b1652615ea9d)

**Review by sirugh — CHANGES_REQUESTED** [review]
Test failed, also some comments.

File: venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js
Comment (dpatil-magento) [reply venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js:56]: We have decided to have assertions also part of functions which makes it easy to maintain in long run, examples can be found here https://github.com/magento/pwa-studio/tree/dev/wishlist-cypress/venia-integration-tests/src/assertions

**Review by revanth0212 — DISMISSED** [review]

**Review by sirugh — APPROVED** [review]
QA Approved, though there is an outstanding TODO that should get addressed :)

File: venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js
Comment (revanth0212) [reply venia-integration-tests/src/tests/wishList/e2eTests/singleWishlistAddRemoveProduct.spec.js:56]: We had a couple of discussions about this in slack and in the Dev time. The decision was to try and omit having assertions as functions but if the assertions are being used too many times and are too many in number, its ok to co-locate them into a function if the developer feels so. 

The important thing is to not overdo it because with multiple developers contributing its hard to keep track of these functions and quickly we can end up with too many assertions and some duplicates.

**Review by revanth0212 — APPROVED** [review]

**Pull request merged**
